chiselFrontend/src/main/scala/chisel3/core/Aggregate.scala

```diff
@@ -226,16 +268,24 @@ sealed class Vec[T <: Data] private (gen: => T, val length: Int)
   override def apply(p: UInt): T = macro CompileOptionsTransform.pArg
 
   def do_apply(p: UInt)(implicit compileOptions: CompileOptions): T = {
-    Binding.checkSynthesizable(p ,s"'p' ($p)")
+    requireIsHardware(p, "vec index")
     val port = gen
-    val i = Vec.truncateIndex(p, length)(UnlocatableSourceInfo, compileOptions)
-    port.setRef(this, i)
 
-    // Bind each element of port to being whatever the base type is
-    // Using the head element as the sample_element
-    for((port_elem, model_elem) <- port.allElements zip sample_element.allElements) {
-      port_elem.binding = model_elem.binding
+    // Reconstruct the resolvedDirection (in Aggregate.bind), since it's not stored.
+    // It may not be exactly equal to that value, but the results are the same.
+    val reconstructedResolvedDirection = direction match {
+      case ActualDirection.Input => UserDirection.Input
+      case ActualDirection.Output => UserDirection.Output
+      case ActualDirection.Bidirectional | ActualDirection.Unspecified => UserDirection.Unspecified
+      case ActualDirection.BidirectionalFlip => UserDirection.Flip
```

Comment:
Done.

Renamed `ActualDirection.Bidirectional` to `ActualDirection.BidirectionalDefault`
Created `ActualDirection.Bidirectional` object that can be matched against (like `case 
ActualDirection.Bidirectional(_)`). It takes an argument of type `ActualDirection.Bidirection.Direction`, so actually filling in the match is a namespace nightmare and I only expect it to be used with `_`. The argument is left in both as a reminder that this can match both default and flipped versions, and allowing extensibility for future versions that might relax the namespace issue.